packages/components/src/snackbar/index.tsx

```diff
@@ -98,7 +98,10 @@ function UnforwardedSnackbar(
 		}, NOTICE_TIMEOUT );
 
 		return () => clearTimeout( timeoutHandle );
-	}, [ onDismiss, onRemove, explicitDismiss ] );
+		// The `onDismiss/onRemove` can have unstable references,
+		// trigger side-effect cleanup, and reset timers.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Disabling exhaustive deps checks is risky because it will prevent ESlint from catching potential bugs for this hook in the future. Could we instead use refs for `onDismiss` and `onRemove` ? Something like:

```js
const onRemoveRef = useRef( onRemove );
useEffect( () => {
  onRemoveRef.current = onRemove;
, [ onRemove ] );

const onDismissRef = useRef( onDismiss );
useEffect( () => {
  onDismissRef.current = onDismiss;
, [ onDismiss ] );

// ..useEffect handling the auto-dismissal timers
```

Reply:
I was just thinking about using the same pattern 😅 I will push updates shortly m.